### PR TITLE
Create new DNS records

### PR DESCRIPTION
**Pull request by szok**

### CODE REVIEW

File: src/ralph/dns/dnsaas.py [hunk 1]
```diff
@@ -82,7 +103,80 @@ def update_dns_records(self, records):
...
+        url = urljoin(
+            settings.DNSAAS_URL, 'api/domains/?'.format(
+                urlencode([('name', domain_name[-1])])
+            )
+        )
+        result = self.get_api_result(url)
+        if result:
+            return result[0]['url']
+
+        return False
```
Comment (mkurek) [inline src/ralph/dns/dnsaas.py:122]: raise exception?

File: src/ralph/dns/dnsaas.py [hunk 2]
```diff
@@ -82,7 +103,80 @@ def update_dns_records(self, records):
...
+        Return domain URL base on record name.
+
+        Args:
+            record: Cleaned data from formset
+
+        Return:
+            Domain URL from API or False if not exists
+        """
+        domain_name = record['name'].split('.', 1)
+        url = urljoin(
```
Comment (mkurek) [inline src/ralph/dns/dnsaas.py:113]: think about caching it

File: src/ralph/dns/dnsaas.py [hunk 3]
```diff
@@ -82,7 +103,80 @@ def update_dns_records(self, records):
...
+        Returns:
+            True or False if an error from api
+        """
+        error = False
+        for item in records:
+            url = urljoin(settings.DNSAAS_URL, 'api/records/')
+            domain = self.get_domain(item)
+            if not domain:
+                logger.error(
+                    'Get domain error from api for record {}'.format(item)
```
Comment (mkurek) [inline src/ralph/dns/dnsaas.py]: Get domain error from api -> Domain not found

File: src/ralph/settings/base.py [hunk 4]
```diff
@@ -291,6 +291,8 @@ def os_env_true(var, default=''):
 DNSAAS_TOKEN = os.environ.get('DNSAAS_TOKEN', '')
 DNSAAS_AUTO_PTR_ALWAYS = os.environ.get('DNSAAS_AUTO_PTR_ALWAYS', 2)
 DNSAAS_AUTO_PTR_NEVER = os.environ.get('DNSAAS_AUTO_PTR_NEVER', 1)
+DNSAAS_OWNER = os.environ.get('DNSAAS_OWNER', '')
```
Comment (mkurek) [inline src/ralph/settings/base.py]: what do you think about default value 'ralph'?

File: src/ralph/dns/views.py [hunk 5]
```diff
@@ -62,6 +64,10 @@ def post(self, request, *args, **kwargs):
                 messages.error(
                     request, _('An error occurred while updating a record')
                 )
+            if to_create and self.dnsaas.create_dns_records(to_create):
```
Comment (mkurek) [inline src/ralph/dns/views.py]: imo we should improve flow of this form - when some of the rows are not saved in dnsaas, there will only be error message at the top. Is it possible to show these rows again with error message near to them?

File: src/ralph/dns/dnsaas.py (hunk 2 above)
Comment (ghost) [inline src/ralph/dns/dnsaas.py:113]: done

**Comment by ghost** [comment]
:ok:

**Pull request merged**